src/plugins/chart_expressions/expression_partition_vis/public/components/visualization_noresults.tsx

```diff
@@ -6,27 +6,37 @@
  * Side Public License, v 1.
  */
 
-import React from 'react';
-import { EuiEmptyPrompt, EuiText } from '@elastic/eui';
-import { i18n } from '@kbn/i18n';
+import React, { FC } from 'react';
+import { EuiIcon, EuiSpacer, EuiText } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { EmptyPlaceholder } from '../../../../charts/public';
+import { ChartTypes } from '../../common/types';
+import { getIcon } from '../utils';
 
-export const VisualizationNoResults = ({ hasNegativeValues = false }) => {
-  return (
-    <EuiEmptyPrompt
-      iconType="visualizeApp"
-      iconColor="default"
-      data-test-subj="pieVisualizationError"
-      body={
-        <EuiText size="xs">
-          {hasNegativeValues
-            ? i18n.translate('expressionPartitionVis.negativeValuesFound', {
-                defaultMessage: "Pie/donut charts can't render with negative values.",
-              })
-            : i18n.translate('expressionPartitionVis.noResultsFoundTitle', {
-                defaultMessage: 'No results found',
-              })}
-        </EuiText>
-      }
-    />
-  );
+interface Props {
+  hasNegativeValues?: boolean;
+  chartType: ChartTypes;
+}
+
+export const VisualizationNoResults: FC<Props> = ({ hasNegativeValues = false, chartType }) => {
+  if (hasNegativeValues) {
+    return (
+      <EuiText
```

Comment:
`EmptyPlaceholder` component takes `icon` and `message` parameters so I think you can replace this code with using it directly. The only problem would be not having `data-test-subj`, but maybe you can add it as a prop to the `EmptyPlaceholder`?

Reply:
@mbondyra, updated the code, according to your proposition. Thanks for the suggestion)